task/source-build/0.1/source-build.yaml

```diff
@@ -0,0 +1,472 @@
+            log.info("Copy source image %s into directory %s", source_image_name, str(base_sources_extraction_dir))
+            try:
+                run(cmd, check=True)
+            except CalledProcessError:
+                log.warning("Failed to copy source image to local %s", source_image_name)
+                return False
+
+            # bsi reads source RPMs from this directory
+            bsi_rpms_dir = create_dir(base_image_sources_dir, "bsi_rpms_dir")
+            sib_dirs.rpm_dir = str(bsi_rpms_dir)  # save this directory for executing bsi
+
+            # bsi reads extra sources from this directory
+            # each source is in its own directory, for instance, subdir/source_a.tar.gz
+            extra_src_dir = create_dir(base_sources_extraction_dir, "extra_src_dir")
+
+            # extract layers, primarily they are RPMs
+            with open(f"{base_sources_extraction_dir}/manifest.json", "r") as f:
+                manifest_data = json.load(f)
+
+            for layer in manifest_data["layers"]:
+                digest = layer["digest"].split(":")[-1]
+                log.debug("untar layer %s", digest)
+
+                blob_member = ""
+                symlink_member = ""
+                with tarfile.open(f"{base_sources_extraction_dir}/{digest}", "r:gz") as tar:
+                    for member in tar:
+                        if member.isfile():
+                            blob_member = member.name
+                        elif member.issym():
+                            symlink_member = member.name
+
+                if symlink_member.startswith("./rpm_dir/"):
+                    dest_dir = sib_dirs.rpm_dir
+                    log.debug("Prepare SRPM %s", symlink_member)
+
+                    extract_blob_member(digest, blob_member, dest_dir,
+                                        rename_to=os.path.basename(symlink_member),
+                                        work_dir=base_sources_extraction_dir,
+                                        log=log)
+
+                elif symlink_member.startswith("./extra_src_dir/"):
+                    extra_src_archive = os.path.basename(symlink_member)
+                    log.debug("Prepare extra source %s", extra_src_archive)
+                    # one extra source archive per directory, no matter what the directory name is.
+                    dest_dir = create_dir(extra_src_dir, extra_src_archive)
+                    sib_dirs.extra_src_dirs.append(dest_dir)  # save this directory for bsi
+
+                    extract_blob_member(digest, blob_member, str(dest_dir),
+                                        rename_to=extra_src_archive,
+                                        work_dir=base_sources_extraction_dir,
+                                        log=log)
+
+                    # FIXME: perhaps the dependency archive (the Cachi2) might be handled differently
+                    run(["tar", "xvf", extra_src_archive], check=True, cwd=dest_dir)
```

Comment:
why are you unpacking all individual remote sources ?

in osbs source image there is "extra_src_dir" and each cachito source archive is in individual layer

Reply:
`extract_blob_member` extracts the tar archive generated by bsi, like `extra_src_0.tar`, then this line extracts the actual source archive, e.g. a remote source in osbs. Then, bsi builds the source image with each of such sources again. As a result, when a user gets the source archive, for example `extra_src_0.tar`, the user is able to get the actual source by unpacking it just once.

Reply:
I'll say it again, you are extracting extra_src_0.tar remote source from osbs source image with cachito remote sources, and ADDing extracted sources to source image, you should be adding that tar, not extracting it

Reply:
oh I see I forgot that extra_src_0.tar is created by bsi and it contains tar.gz from cachito

Reply:
it will work also fro cachito2 sources, because you will extract the archive from cachito2 and get that directory structure with archive, and bsi will again make extra_src.tar from it, so it works the same